src/app.js

```diff
@@ -1,7 +1,20 @@
 /* @flow */
 'use strict'
 
+// SIGINT and uncaughtException do not run cleanup
+// so we have an exitHandler to clean up the database connection
+function exitHandler (error) {
+  if (error) {
+    console.error('Exiting: ', error)
+    process.exit(1)
+  }
+  process.exit(0)
+}
+
 module.exports = require('./lib/app')
 if (!module.parent) {
+  process.on('SIGINT', exitHandler)
+  process.on('uncaughtException', exitHandler)
```

Comment:
> Knex maintains a connection pool, which does not close automatically on a SIGINT.

Oh, so just call `knex.destroy()`.

If you don't call `process.exit()`, then you're not preventing the default handler you're just adding something to it. That should be pretty safe for both on-exit and on-uncaughtException. (They're [just standard EventEmitters](https://github.com/nodejs/node/blob/4d4cfb27ca7718c7df381ac3b257175927cd17d1/lib/internal/bootstrap_node.js#L268), so as long as your listener doesn't throw and doesn't `process.exit`, they'll do their thing after it returns.)